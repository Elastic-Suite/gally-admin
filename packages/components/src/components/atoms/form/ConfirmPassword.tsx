import React from 'react'

import { useTranslation } from 'next-i18next'

import InputText from './InputText'

export interface IConfirmPasswordValues {
  password: string | null
  confirmPassword: string | null
}

export interface IConfirmPasswordProps {
  value?: IConfirmPasswordValues
  passwordLabel?: string
  confirmPasswordLabel?: string
  onChange?: (values: IConfirmPasswordValues) => void
  showError?: boolean
}

function ConfirmPassword(props: IConfirmPasswordProps): JSX.Element {
  const { value, passwordLabel, confirmPasswordLabel, onChange, showError } =
    props
  const { t } = useTranslation('common')

  function onChangePassword(password: string): void {
    onChange({ ...value, password })
  }

  function onChangeConfirmPassword(confirmPassword: string): void {
    onChange({ ...value, confirmPassword })
  }

  return (
    <>
      <InputText
        fullWidth
        label={t(passwordLabel ?? 'newPassword.label')}
        margin="normal"
        onChange={onChangePassword}
        type="password"
        value={value?.password}
        showError={showError}
        required
      />
      <InputText
        fullWidth
        label={t(confirmPasswordLabel ?? 'confirmPassword.label')}
        margin="normal"
        onChange={onChangeConfirmPassword}
        type="password"
        value={value?.confirmPassword}
        showError={showError}
        required
        additionalValidator={(valueToValidate: string): string => {
          return valueToValidate !== value?.password ? 'mismatchPassword' : ''
        }}
      />
    </>
  )
}

export default ConfirmPassword
