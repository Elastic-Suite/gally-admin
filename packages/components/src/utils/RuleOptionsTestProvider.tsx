import React, { ReactNode } from 'react'
import {
  IRuleEngineOperators,
  RuleAttributeType,
  getIri,
} from '@elastic-suite/gally-admin-shared'

import ruleEngineOperator from '../../public/mocks/rule_engine_operators.json'
import sourceFields from '../../public/mocks/source_fields.json'
import sourceFieldLabels from '../../public/mocks/source_field_labels.json'

import RuleOptionsProvider from '../components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider'

interface IProps {
  children: ReactNode
}

const sourceFieldLabelsMap = new Map(
  sourceFieldLabels['hydra:member'].map((label) => [
    label.sourceField,
    label.label,
  ])
)

const fields = sourceFields['hydra:member'].map((field) => {
  const label = sourceFieldLabelsMap.get(getIri('source_fields', field.id))
  return {
    id: field.id,
    code: field.code,
    label: (label || field.defaultLabel) ?? '',
    type: field.type as RuleAttributeType,
  }
})

function RuleOptionsTestProvider(props: IProps): JSX.Element {
  const { children } = props
  return (
    <RuleOptionsProvider
      catalogId={-1}
      localizedCatalogId={-1}
      fields={fields}
      ruleOperators={ruleEngineOperator as IRuleEngineOperators}
    >
      {children}
    </RuleOptionsProvider>
  )
}

export default RuleOptionsTestProvider
