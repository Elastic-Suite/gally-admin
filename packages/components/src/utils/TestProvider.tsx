import React, { ReactNode } from 'react'

import catalogs from '../../public/mocks/catalog.json'
import metadata from '../../public/mocks/metadata.json'

import { catalogContext, metadataContext } from '../contexts'
import { useCatalogs } from '../hooks'

import OptionsProvider from '../components/stateful-providers/OptionsProvider/OptionsProvider'

interface IProps {
  children: ReactNode
}

function TestProvider(props: IProps): JSX.Element {
  const { children } = props
  const contextValueCatalog = useCatalogs(catalogs['hydra:member'])
  const contextValueMetadata = metadata['hydra:member']

  return (
    <OptionsProvider>
      <catalogContext.Provider value={contextValueCatalog}>
        <metadataContext.Provider value={contextValueMetadata}>
          {children}
        </metadataContext.Provider>
      </catalogContext.Provider>
    </OptionsProvider>
  )
}

export default TestProvider
