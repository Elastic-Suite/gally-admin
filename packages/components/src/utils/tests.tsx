import React, { ReactNode } from 'react'
import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { Bundle, api } from '@elastic-suite/gally-admin-shared'

import { AppStore, RootState, setupStore } from '../store'

import AppProvider from '../components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: {
      'base_url/media': 'https://localhost/media/catalog/product/',
    },
    metadata: [
      {
        '@id': '/metadata/3',
        '@type': 'Metadata',
        id: 3,
        entity: 'product',
        sourceFields: ['/source_fields/217', '/source_fields/253'],
      },
      {
        '@id': '/metadata/4',
        '@type': 'Metadata',
        id: 4,
        entity: 'category',
        sourceFields: ['/source_fields/254', '/source_fields/286'],
      },
    ],
  },
  user: {
    requestedPath: '',
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
