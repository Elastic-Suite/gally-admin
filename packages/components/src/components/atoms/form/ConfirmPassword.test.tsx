import React from 'react'

import { renderWithProviders } from '../../../utils/tests'

import ConfirmPassword from './ConfirmPassword'
import { screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

describe('ConfirmPassword', () => {
  it('should display initial state', () => {
    const { container } = renderWithProviders(<ConfirmPassword value={null} />)
    expect(container).toMatchSnapshot()
  })

  it('should display with different labels', () => {
    const { container } = renderWithProviders(
      <ConfirmPassword
        value={null}
        passwordLabel="Other password label"
        confirmPasswordLabel="Other confirm password label"
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should not display error with showError prop when the password are the same', () => {
    const { container } = renderWithProviders(
      <ConfirmPassword
        value={{ password: 'aPassword', confirmPassword: 'aPassword' }}
        showError
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should display error with showError prop when the fields are required with missing values', () => {
    renderWithProviders(
      <ConfirmPassword
        value={{ password: '', confirmPassword: '' }}
        showError
      />
    )
    expect(screen.getAllByText('formError.valueMissing')).toHaveLength(2)
  })

  it('should display error with showError prop when the password are not the same', () => {
    renderWithProviders(
      <ConfirmPassword
        value={{ password: 'aPassword', confirmPassword: 'aConfirmPassword' }}
        showError
      />
    )
    expect(screen.getByText('formError.mismatchPassword')).toBeInTheDocument()
  })
})
