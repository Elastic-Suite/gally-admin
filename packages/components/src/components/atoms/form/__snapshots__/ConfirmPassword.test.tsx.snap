// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmPassword should display initial state 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth css-sg07kc-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required css-rnc4nv-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      newPassword.label
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ufxisk-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      class="InputText__Wrapper css-17fp4jk"
    >
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-33c5sc-MuiInputBase-root"
        required=""
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
          data-testid="inputText"
          required=""
          type="password"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth css-sg07kc-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required css-rnc4nv-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      confirmPassword.label
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ufxisk-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      class="InputText__Wrapper css-17fp4jk"
    >
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-33c5sc-MuiInputBase-root"
        required=""
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
          data-testid="inputText"
          required=""
          type="password"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ConfirmPassword should display with different labels 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth css-sg07kc-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required css-rnc4nv-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Other password label
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ufxisk-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      class="InputText__Wrapper css-17fp4jk"
    >
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-33c5sc-MuiInputBase-root"
        required=""
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
          data-testid="inputText"
          required=""
          type="password"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth css-sg07kc-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required css-rnc4nv-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Other confirm password label
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ufxisk-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      class="InputText__Wrapper css-17fp4jk"
    >
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-33c5sc-MuiInputBase-root"
        required=""
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
          data-testid="inputText"
          required=""
          type="password"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ConfirmPassword should not display error with showError prop when the password are the same 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth css-sg07kc-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-rnc4nv-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      newPassword.label
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ufxisk-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      class="InputText__Wrapper css-17fp4jk"
    >
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1jf161a-MuiInputBase-root"
        required=""
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
          data-testid="inputText"
          required=""
          type="password"
          value="aPassword"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth css-sg07kc-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-rnc4nv-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      confirmPassword.label
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ufxisk-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      class="InputText__Wrapper css-17fp4jk"
    >
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1jf161a-MuiInputBase-root"
        required=""
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
          data-testid="inputText"
          required=""
          type="password"
          value="aPassword"
        />
      </div>
    </div>
  </div>
</div>
`;
